elasticdl/python/worker/worker.py

```diff
@@ -559,47 +563,50 @@ def run(self):
         # Initialize `local_update_count=get_model_steps` in order to set
         # `train_with_local_model` to False inside for-loop for the first
         # minibatch.
+
         local_update_count = self._get_model_steps
         last_training_minibatch_failed = False
-        last_minibatch_do_evaluation = False
         while True:
             dataset = self._task_data_service.get_dataset()
             if not dataset:
                 break
             dataset = self._dataset_fn(
-                dataset, mode, self._task_data_service.data_reader.metadata
+                dataset,
+                Mode.TRAINING,
+                self._task_data_service.data_reader.metadata,
             )
             dataset = dataset.batch(self._minibatch_size).prefetch(1)
             for dataset_batch in dataset:
                 if self._job_type == JobType.TRAINING_WITH_EVALUATION:
-                    if self._process_eval_task_if_needed():
```

Comment:
Please leave this logic, because if a worker does some evaluation tasks in the training process, it should pull model instead of training with local model parameters.

Reply:
Yeah, we can get a status from `_evaluate_only`.